instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/TripSessionsBillingTest.kt

```diff
@@ -0,0 +1,316 @@
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.sdkTest
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.setNavigationRoutesAndWaitForUpdate
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.startTripSessionAndWaitForActiveGuidanceState
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.startTripSessionAndWaitForFreeDriveState
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.stopTripSessionAndWaitForIdleState
+import com.mapbox.navigation.instrumentation_tests.utils.http.HttpServiceEvent
+import com.mapbox.navigation.instrumentation_tests.utils.http.HttpServiceEventsObserver
+import com.mapbox.navigation.instrumentation_tests.utils.parameters
+import com.mapbox.navigation.instrumentation_tests.utils.routes.RoutesProvider
+import com.mapbox.navigation.instrumentation_tests.utils.routes.RoutesProvider.toNavigationRoutes
+import com.mapbox.navigation.testing.ui.BaseCoreNoCleanUpTest
+import com.mapbox.navigation.testing.ui.utils.getMapboxAccessTokenFromResources
+import com.mapbox.navigation.testing.ui.utils.runOnMainSync
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.take
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.net.URI
+import java.util.concurrent.TimeUnit
+
+class TripSessionsBillingTest : BaseCoreNoCleanUpTest() {
+
+    @get:Rule
+    val mapboxNavigationRule = MapboxNavigationRule()
+
+    private lateinit var mapboxNavigation: MapboxNavigation
+
+    private val route = RoutesProvider.dc_very_short(context).toNavigationRoutes()[0]
+
+    private lateinit var httpEventsObserver: HttpServiceEventsObserver
+
+    override fun setupMockLocation(): Location = mockLocationUpdatesRule.generateLocationUpdate {
+        route.routeOptions.coordinatesList().first().run {
+            latitude = latitude()
+            longitude = longitude()
+        }
+        bearing = 190f
+    }
+
+    @Before
+    fun setup() {
+        httpEventsObserver = HttpServiceEventsObserver.register()
+
+        runOnMainSync {
+            mapboxNavigation = MapboxNavigationProvider.create(
+                NavigationOptions.Builder(context)
+                    .accessToken(getMapboxAccessTokenFromResources(context))
+                    .routingTilesOptions(
+                        RoutingTilesOptions.Builder()
+                            .tilesBaseUri(URI(mockWebServerRule.baseUrl))
+                            .build()
+                    )
+                    .build()
+            )
+        }
+    }
+
+    @After
+    fun cleanUp() {
+        HttpServiceFactory.getInstance().setInterceptor(null)
```

Comment:
NIT: it would be better to invoke an `unregister` method here and nullify the interceptor there. 
Otherwise we use register not actually caring what's happening there but in clean-up we act like we know what's been done in `register`.